code/game/objects/items/weapons/implants/implantreagent_vr.dm

```diff
@@ -96,3 +96,90 @@
+	desc = "This is an implant that has attached storage and generates a scanned reagent."
+	gen_amount = 0.2 //This is extremely powerful and, as such, has a long time to create reagents
+	gen_cost = 10 //And an extremely high cost to make reagents. 1000 nutrition to go from nothing from full in 100 ticks.
+	transfer_amount = 20
+	usable_volume = 20 //Can't hold much, either.
+	generated_reagent = "water" //Placeholder reagent until the person selects a new reagent.
+	var/assigned_proc2 = /mob/living/carbon/human/proc/scan_reagent_with_implant
+	var/verb_name2 = "Scan reagent with implant"
+	var/verb_desc2 = "Scans a reagent in a container and begins to synthetsise it."
+
+
+/obj/item/weapon/implant/reagent_generator/scannable/implanted(mob/living/carbon/source)
+	processing_objects += src
+	to_chat(source, "<span class='notice'>You implant [source] with \the [src].</span>")
+	assigned_proc = new assigned_proc(source, verb_name, verb_desc)
+	assigned_proc2 = new assigned_proc2(source, verb_name2, verb_desc2)
+	return 1
+
+/obj/item/weapon/implant/reagent_generator/scannable/process()
+	var/before_gen
+	if(imp_in && generated_reagent)
+		before_gen = reagents.total_volume
+		if(reagents.total_volume < reagents.maximum_volume)
+			if(imp_in.nutrition >= gen_cost)
+				do_generation()
+		else
+			return
+	else
+		imp_in.verbs -= assigned_proc
+		imp_in.verbs -= assigned_proc2
+		return
+
+	if(reagents)
+		if(reagents.total_volume == reagents.maximum_volume * 0.05)
+			to_chat(imp_in, "<span class='notice'>[pick(empty_message)]</span>")
+		else if(reagents.total_volume == reagents.maximum_volume && before_gen < reagents.maximum_volume)
+			to_chat(imp_in, "<span class='warning'>[pick(full_message)]</span>")
+
+/mob/living/carbon/human/proc/scan_reagent_with_implant()
+	set name = "Scan reagent and begin synthesis"
+	set desc = "Scan a container and begin internal synthesis of the reagent that is the majority of the container."
+	set category = "Object"
+	set src in view(1)
+
+	do_reagent_scan(usr)
+
+
+
+/mob/living/carbon/human/proc/do_reagent_scan(var/mob/living/carbon/human/user = usr)
+	if(!isliving(user) || !user.canClick())
+		return
+
+	if(user.incapacitated() || user.stat > CONSCIOUS)
+		return
+
+	var/obj/item/weapon/reagent_containers/container = user.get_active_hand()
+	if(!container)
+		to_chat(user,"<span class='notice'>You need an open container in your hand to do this!</span>")
+		return
+
+
+	var/obj/item/weapon/implant/reagent_generator/scannable/rimplant
+	for(var/I in src.contents)
+		if(istype(I, /obj/item/weapon/implant/reagent_generator/scannable))
+			rimplant = I
+			break
+	if(rimplant)
+		if(!container.reagents) //This should only apply to things that can't hold reagents.
+			to_chat(user,"<span class='notice'>The container must have one single, pure, liquid reagent inside of it to scan.</span>")
+			return
+		var/datum/reagent/R = container.reagents.get_master_reagent()
+		if(R.reagent_state != 2)
```

Comment:
Use the constants used elsewhere and not just the number please. I have no idea what '2' is.